src/target/riscv/riscv-013.c

```diff
@@ -1754,7 +1761,7 @@ static int examine(struct target *target)
 
 	if (target->smp) {
 		bool haltgroup_supported;
-		if (set_haltgroup(target, &haltgroup_supported) != ERROR_OK)
+		if (set_group(target, &haltgroup_supported, target->smp, HALTGROUP) != ERROR_OK)
```

Comment:
Broader point that goes beyond this merge request: 

What is please the meaning of `target->smp`? Is it a bool, denoting if SMP is enabled for that target? Or is it an integer, containing the number of the SMP cluster? 

It seems those two approaches are mixed:
* OpenOCD (outside of the riscv target) code apparently treats target->smp just as a bool.
* On the other hand, the riscv target expects takes it directly as the cluster group --> halt group number in this function.

And in this piece of code, there is apparently a merge mistake (target->smp is always set to 1):
https://github.com/riscv/riscv-openocd/blob/riscv/src/target/target.c#L6473

My recommendation would be to make target->smp a real bool. For the SMP cluster number, another struct member could be introduced (`target->smp_group_num`). Preferably, this should be submitted also upstream to have the SMP handled consistently.

Reply:
I added this in https://github.com/riscv/riscv-openocd/pull/353. It's not yet upstreamed, so no other targets do anythign with it.